lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
@NullVoxPopuli from jsonapi.org:

> The value of the relationships key MUST be an object (a “relationships object”)

That tells me, can't be null. If it can be null, this seems reasonable.

**data** being null or [] is the only no go.